datalad_next/commands/status.py

```diff
@@ -0,0 +1,212 @@
+            reading configuration items, this command does not interact with
+            the dataset."""),
+        untracked=Parameter(
+            args=('--untracked',),
+            nargs='?',
+            const='no-empty-dir',
+            doc="""If and how untracked content is reported when comparing
+            a revision to the state of the working tree. 'no': no untracked
+            content is reported; 'normal': untracked files and entire
+            untracked directories are reported as such; 'all': report
+            individual files even in fully untracked directories."""),
+        recursive=Parameter(
+            args=('-r', '--recursive'),
+            nargs='?',
+            const='datasets',
+            doc="some"),
+    )
+
+    _examples_ = [
+    ]
+
+    @staticmethod
+    @datasetmethod(name="next_status")
+    @eval_results
+    def __call__(
+        # TODO later
+        #path=None,
+        *,
+        dataset=None,
+        # TODO later
+        #annex=None,
+        untracked='whole-dir',
+        recursive='repository',
+        # TODO unclear if needed
+        #eval_subdataset_state='full',
+    ) -> Generator[StatusResult, None, None] | list[StatusResult]:
+        ds = dataset.ds
+        rootpath = Path.cwd() if dataset.original is None else ds.pathobj
+
+        for item in iter_gitstatus(
+            path=rootpath,
+            untracked=untracked,
+            recursive=recursive,
+        ):
+            yield StatusResult(
+                action='status',
+                status=CommandResultStatus.ok,
+                path=rootpath / (item.path or item.prev_path),
+                type=item.gittype,
+                prev_type=item.prev_gittype,
+                diff_state=item.status,
+                refds=ds,
+                logger=lgr,
+            )
+
+    def custom_result_renderer(res, **kwargs):
+        # we are guaranteed to have dataset-arg info through uniform
+        # parameter validation
+        dsarg = kwargs['dataset']
+        rootpath = Path.cwd() if dsarg.original is None else dsarg.ds.pathobj
+        # because we can always determine the root path of the command
+        # execution environment, we can report meaningful relative paths
+        # unconditionally
+        path = res.path.relative_to(rootpath)
+        # collapse item type information across current and previous states
+        type_ = res.type or res.prev_type
+        type_str = '' if type_ is None else type_.value
+        max_len = len('untracked')
+        # message format is same as for previous command implementation
+        ui.message(u'{fill}{state}: {path}{type_}'.format(
+            fill=' ' * max(0, max_len - len(res.state.value)),
+            state=ac.color_word(
+                res.state.value,
+                STATE_COLOR_MAP.get(res.state)),
+            path=path,
+            type_=' ({})'.format(ac.color_word(type_str, ac.MAGENTA))
+            if type_str else ''),
```

Comment:
I noticed that unlike ``datalad status``, ``next-status`` does not report "nothing to save, working tree clean" when run in a clean dataset. Is that intentional?

Reply:
I have not ported the result summary renderer yet. No other reason.

Reply:
Done now